### DISCUSSION TITLE
Example usage for FormatConverter

### DISCUSSION THREAD

**Discussion by GuillaumeLeclerc** [pschatzmann/arduino-audio-tools#515] [Q&A] [answered]
Hello,

I need to convert from 16bits to 32bits and the FormatConverter class which seem to be the way to do it doesn't extend BaseConverter so it's unusable in a pipeline. What is the best way to use it ?

Thanks

**Comment by pschatzmann** [comment, marked as answer]
You can use one of the following classes to convert between integer data types 
- NumberFormatConverterStream or 
- NumberFormatConverterStreamT< TFrom, TTo > Class
- FormatConverterStream

Check out the class documentation under Transformation: https://pschatzmann.github.io/arduino-audio-tools/html/modules.html
Here is an [example](https://github.com/pschatzmann/arduino-audio-tools/blob/main/jupyter/Tests.ipynb) that shows how to convert the data type.

ps. I will delete the (obsolete) FormatConverter to avoid any future confusion and maybe I should change the title from Transformation to something else (e.g. Data Conversion). Would you have any suggestions ?
